nisapi/__init__.py

```diff
@@ -93,65 +93,90 @@ def _get_dataset_ids() -> Sequence[str]:
 
 def _cache_clean_dataset(
     id: str,
-    root_path: Path,
+    cache_path: Path,
     app_token: Optional[str],
     overwrite: str,
     validation_mode: str,
 ) -> None:
-    raw_data = _get_nis_raw(id, root_path=root_path, app_token=app_token)
+    raw_data_path = get_data_path(path=cache_path, type_="raw", id=id)
+    raw_data = _get_nis_raw(id=id, data_path=raw_data_path, app_token=app_token)
     clean_data = nisapi.clean.clean_dataset(
         df=raw_data, id=id, validation_mode=validation_mode
     )
-    clean_path_dir = _dataset_cache_path(root_path=root_path, type_="clean", id=id)
-    clean_path = clean_path_dir / "part-0.parquet"
+    clean_data_path = get_data_path(path=cache_path, type_="clean", id=id)
+    clean_data_filepath = clean_data_path / "part-0.parquet"
 
-    if clean_path.exists():
-        msg = f"Clean dataset {clean_path} already exists"
+    if clean_data_filepath.exists():
+        msg = f"Clean dataset {clean_data_filepath} already exists"
         if overwrite == "warn":
             warnings.warn(msg)
             return None
         else:
             raise RuntimeError(f"Invalid overwrite option '{overwrite}'")
 
-    if not clean_path_dir.exists():
-        clean_path_dir.mkdir(parents=True)
+    if not clean_data_path.exists():
+        clean_data_path.mkdir(parents=True)
 
-    clean_data.write_parquet(clean_path)
+    clean_data.write_parquet(clean_data_filepath)
 
 
-def _root_cache_path() -> Path:
-    return Path(platformdirs.user_cache_dir("nisapi"))
+def get_data_path(
+    path: Optional[Path] = None,
+    type_: str = "clean",
```

Comment:
```suggestion
    type_: Literal["clean", "raw"] = "clean",
```

might be cleaner, with a `from typing import Literal`